src/libcollections/vec.rs

```diff
@@ -1756,6 +1803,122 @@ impl<'a, T> Drop for Drain<'a, T> {
     }
 }
 
-
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<'a, T> ExactSizeIterator for Drain<'a, T> {}
+
+/// A splicing iterator for `Vec<T>`. See the [`Vec::splice`](struct.Vec.html#method.splice) method.
+#[unstable(feature = "splice", reason = "recently added", issue = "32310")]
+pub struct Splice<'a, I: Iterator + 'a> {
+    drain: Drain<'a, I::Item>,
+    replace_with: I,
+}
+
+#[unstable(feature = "splice", reason = "recently added", issue = "32310")]
+impl<'a, I: Iterator> Iterator for Splice<'a, I> {
+    type Item = I::Item;
+
+    #[inline]
```

Comment:
It should be safe to elide these `#[inline]` annotations because this impl is already generic (and hence monomorphized)